src/cmi_docx/comment.py

```diff
@@ -135,3 +137,249 @@ def add_comment(
     comment_reference.append(comment_reference_run)
 
     elements[0].append(comment_reference)
+
+
+@dataclass
+class CommentRange:
+    """All data representing a comment range.
+
+    Used for temporal removal and reinsertion via `CommentPreserver`.
+    """
+
+    id: str
+    start_index: int
+    end_index: int
+
+
+class CommentPreserver:
+    """Utility to remove and reinsert comment ranges.
+
+    As some document modifications will affect comment ranges
+    in weird ways this allows to preserve them.
+    """
+
+    def __init__(self, paragraph_element: etree._Element) -> None:
+        """Create CommentPreserver utility."""
+        self.paragraph = paragraph_element
+        self.ns = {"w": "http://schemas.openxmlformats.org/wordprocessingml/2006/main"}
+
+    def _get_text_length_before_element(self, element: etree._Element) -> int:
+        """Calculate total text length before given element."""
+        length = 0
+        for elem in self.paragraph.iter():
+            if elem == element:
+                break
+            if elem.tag == f"{{{self.ns['w']}}}t":
+                length += len(elem.text or "")
+        return length
+
+    def extract_comments(self) -> list[CommentRange]:
+        """Extract all comment ranges from the paragraph."""
+        comments = []
+        start_elements = {}
+
+        # First pass: collect all comment range starts
+        for elem in self.paragraph.iter():
+            if elem.tag == f"{{{self.ns['w']}}}commentRangeStart":
+                comment_id = elem.get(f"{{{self.ns['w']}}}id")
+                start_pos = self._get_text_length_before_element(elem)
+                start_elements[comment_id] = start_pos
+
+        # Second pass: match with ends
+        for elem in self.paragraph.iter():
+            if elem.tag == f"{{{self.ns['w']}}}commentRangeEnd":
+                comment_id = elem.get(f"{{{self.ns['w']}}}id")
+                if comment_id in start_elements:
+                    end_pos = self._get_text_length_before_element(elem)
+                    comments.append(
+                        CommentRange(
+                            id=comment_id,
+                            start_index=start_elements[comment_id],
+                            end_index=end_pos,
+                        )
+                    )
+
+        return comments
+
+    def strip_comments(self) -> None:
+        """Remove all comment-related elements from the paragraph."""
+        to_remove = []
+
+        def should_remove_run(run) -> bool:  # noqa: ANN001 # Run is unspecified internal type of docx-python
+            """Helper to determine if a run should be removed."""
+            # Check for nested runs
```

Comment:
Nested runs exist!?

Reply:
Yeah I dont think they should but docx-python sometimes seems to create them